pkg/operator/workload/manage.go

```diff
@@ -0,0 +1,288 @@
+package workload
```

Comment:
The PR description mentions moving https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/sync_openshiftcontrollermanager_v311_00.go to this new package, but I'm a little unclear on what exactly the change gains us.

Can you elaborate on the motivations @otaviof ?  Of if this was more than simply moving code from https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/sync_openshiftcontrollermanager_v311_00.go and I'm missing what was added, can you highlight that either by PR comment or code comment ?

thanks

Reply:
@otaviof this comment still needs to be addressed.

Reply:
> The PR description mentions moving https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/sync_openshiftcontrollermanager_v311_00.go to this new package, but I'm a little unclear on what exactly the change gains us.

By moving `sync_openshiftcontrollermanager_v311_00.go` to a standalone package we gain isolation of the primary logic of this operator in a single place. And based upon other operators OpenShift organization I see that "workload" is a common name for this type of package.

We should refactor the `workload` package further, possibly adopting  more of library-go's `NewResourceSyncController`, in subsequent PRs.

> Can you elaborate on the motivations @otaviof ? Of if this was more than simply moving code from https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/sync_openshiftcontrollermanager_v311_00.go and I'm missing what was added, can you highlight that either by PR comment or code comment ?

In this PR we've improved the unit and integration tests, plus refactored the routes into a new package. By doing so the code was reorganized into smaller/simpler functions, all doc-comments have been reviewed and such.

What's your take on this, @gabemontero?